hydromt_wflow/workflows/rootzoneclim.py

```diff
@@ -0,0 +1,875 @@
+            ) 
+        precip_effective = np.zeros(
+            (len(ds_sub.sel(forcing_type=forcing_type)["index"]), 
+             nr_time_steps)
+            ) 
+        canopy_storage = np.zeros(
+            (len(ds_sub.sel(forcing_type=forcing_type)["index"]), 
+             nr_time_steps)
+            )
+       
+        # Load the potential evaporation and precipitation into memory to speed up
+        # the subsequent for loop.
+        #make sure order of coord is the same. 
+        Epdt = ds_sub.sel(forcing_type=forcing_type)["pet_mean"].dropna("time").transpose("index", "time").values
+        Pdt = ds_sub.sel(forcing_type=forcing_type)["precip_mean"].dropna("time").transpose("index", "time").values
+        # Loop through the time steps and determine the variables per time step.
+        for i in range(0, nr_time_steps):
+            if intercep_vars_sub != None:
+                #TODO: for now assumed that LAI contains monthly data, change this for future
+                month = pd.to_datetime(ds_sub.time[i].values).month
+                Imax = intercep_vars_sub["Imax"].sel(time=month)
+            # Determine the variables with a simple interception reservoir approach
+            canopy_storage[:,i] = canopy_storage[:,i] + Pdt[:,i]
+            precip_effective[:,i] = np.maximum(0, canopy_storage[:,i] - Imax)
+            canopy_storage[:,i] = canopy_storage[:,i] - precip_effective[:,i]
+            evap_interception[:,i] = np.minimum(Epdt[:,i], canopy_storage[:,i])
+            canopy_storage[:,i] = canopy_storage[:,i] - evap_interception[:,i]
+
+            # Update Si for the next time step
+            if i < nr_time_steps - 1:
+                canopy_storage[:,i+1] = canopy_storage[:,i]
+        
+        #insert in ds for the time that is available in each forcing type for precip and pet
+        time_forcing_type = ds_sub[["precip_mean", "pet_mean"]].sel(forcing_type=forcing_type).dropna("time").time
+        ds_sub["evap_interception"].loc[dict(forcing_type=forcing_type, time = time_forcing_type)] = evap_interception
+        ds_sub["precip_effective"].loc[dict(forcing_type=forcing_type, time = time_forcing_type)] = precip_effective
+        ds_sub["canopy_storage"].loc[dict(forcing_type=forcing_type, time = time_forcing_type)] = canopy_storage    
+    
+    return ds_sub
+
+
+def determine_storage_deficit(ds_sub, correct_cc_deficit):
+    """
+    Function to determine the storage deficit for every time step, subcatchment
+    location and dataset in ds_sub.
+
+    Parameters
+    ----------
+    ds_sub : xr.Dataset
+        Dataset containing the daily or higher-resolution data.
+
+    Returns
+    -------
+    ds_sub : xr.Dataset
+        Same as above, but containing the storage deficits per time step for all
+        forcing types.
+    """         
+    #make sure the order of the coordinates is always the same. 
+    # Calculate it per forcing type
+
+    ds_sub["storage_deficit"] = (
```

Comment:
You could think of using hydromt.raster.full_like method instead (also in other places like this). Advantage is that you can choose to keep the new array lazy and np.nan can be directly set as the data values

Reply:
good point, I have replaced this here and at another point in the code!